File: Backend/TrackrAPI/TrackrAPI/Services/GestionExamen/RespuestasClasificacionPreguntaService.cs


using TrackrAPI.Dtos.GestionExamen;
using TrackrAPI.Models;
using TrackrAPI.Repositorys.GestionExamen;

namespace TrackrAPI.Services.GestionExamen;
public class RespuestasClasificacionPreguntaService
{

    private readonly IRespuestasClasificacionPreguntaRepository _respuestasClasificacionPreguntaRepository;

    public RespuestasClasificacionPreguntaService(IRespuestasClasificacionPreguntaRepository respuestasClasificacionPreguntaRepository)
    {
        _respuestasClasificacionPreguntaRepository = respuestasClasificacionPreguntaRepository;
    }
    public IEnumerable<RespuestasClasificacionPreguntaGridDto> ConsultarParaGrid(int idClasificacionPregunta)
    {
        return _respuestasClasificacionPreguntaRepository.ConsultarParaGrid(idClasificacionPregunta);
    }
    public RespuestasClasificacionPreguntaInformacionGeneralDto ConsultarParaFormulario(int idRespuestasClasificacionPregunta)
    {
        return _respuestasClasificacionPreguntaRepository.ConsultarParaFormulario(idRespuestasClasificacionPregunta);
    }
    public void Agregar(RespuestasClasificacionPreguntaFormularioDto captura)
    {
        try
        {
            var rcp = new RespuestasClasificacionPregunta
            {
                IdRespuestasClasificacionPregunta = captura.IdRespuestasClasificacionPregunta,
                IdClasificacionPregunta = captura.IdClasificacionPregunta,
                Nombre = captura.Nombre,
                Valor = captura.Valor,
                Estatus = true,
                Identificador = captura.Identificador
            };
            _respuestasClasificacionPreguntaRepository.Agregar(rcp);

        }
        catch (Exception ex)
        {
            Console.WriteLine(ex.Message);
        }
    }

    public void Editar(RespuestasClasificacionPreguntaFormularioDto captura)
    {
        try
        {
            var rcp = new RespuestasClasificacionPregunta
            {
                IdRespuestasClasificacionPregunta = captura.IdRespuestasClasificacionPregunta,
                IdClasificacionPregunta = captura.IdClasificacionPregunta,
                Nombre = captura.Nombre,
                Valor=captura.Valor,
                Estatus = captura.Estatus,
                Identificador = captura.Identificador
            };
            _respuestasClasificacionPreguntaRepository.Editar(rcp);

        }
        catch (Exception ex)
        {
            Console.WriteLine(ex.Message);
        }
    }

    public void Eliminar(int idRespuestasClasificacionPregunta)
    {
        _respuestasClasificacionPreguntaRepository.Eliminar(idRespuestasClasificacionPregunta);
    }
}

File: Frontend/trackr-administrador/src/app/shared/http/examen/respuestas-clasificacion-pregunta.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { RespuestasClasificacionPreguntaGridDto } from '../../examen/respuestas-clasificacion-pregunta-grid-dto';
import { RespuestasClasificacionPreguntaInformacionGeneralDto } from '../../examen/respuestas-clasificacion-pregunta-informacion-general-dto';
import { RespuestasClasificacionPreguntaFormularioDto } from '../../examen/respuestas-clasificacion-pregunta-formulario-dto';

@Injectable({
  providedIn: 'root'
})
export class RespuestasClasificacionPreguntaService {
  private dataUrl = 'respuestasClasificacionPregunta/';
  constructor(private http: HttpClient) {}

  consultarParaGrid(idClasificacionPregunta: number): Observable<RespuestasClasificacionPreguntaGridDto[]> {
    return this.http.get<RespuestasClasificacionPreguntaGridDto[]>(this.dataUrl+`grid/${idClasificacionPregunta}`);
  }
  
  consultarParaFormulario(idRespuestasClasificacionPregunta: number): Observable<RespuestasClasificacionPreguntaInformacionGeneralDto> {
    return this.http.get<RespuestasClasificacionPreguntaInformacionGeneralDto>(this.dataUrl+`${idRespuestasClasificacionPregunta}`);
  }
  agregar(captura: RespuestasClasificacionPreguntaFormularioDto): Observable<void> {
    return this.http.post<void>(this.dataUrl+`agregar`, captura);
  }
  

  editar(captura: RespuestasClasificacionPreguntaFormularioDto): Observable<void>{
    return this.http.put<void>(this.dataUrl+`editar`,captura);
  }

  public eliminar(idRespuestasClasificacionPregunta: number): Observable<void> {
    return this.http.delete<void>(this.dataUrl + `${idRespuestasClasificacionPregunta}`);
  }
}
